File: python/all-your-base/all_your_base.py
def rebase(from_base, digits, to_base):
    if from_base < 2 or to_base < 2:
        raise ValueError('Bases must be at least 2.')

    if any(map(lambda digit: not 0 <= digit < from_base, digits)):
        raise ValueError('Digits are out of bound.')

    number = 0
    for digit in digits:
        number = number * from_base + digit

    target_digits = []
    while number:
        target_digits.append(number % to_base)
        number //= to_base

    if not target_digits:
        target_digits.append(0)

    return target_digits[::-1]

File: cpp/all-your-base/all_your_base.cpp
#include "all_your_base.h"

namespace all_your_base
{
std::vector<unsigned int> convert(int from_base, const std::vector<unsigned int> &digits, int to_base)
{
    if (from_base < 2 || to_base < 2)
    {
        throw std::invalid_argument{"Bases must be at least 2."};
    }

    if (!std::all_of(digits.begin(), digits.end(), [&](int digit) { return digit >= 0 && digit < from_base; }))
    {
        throw std::invalid_argument{"Digits are out of bound."};
    }

    int number{0};
    for (int digit : digits)
    {
        number = number * from_base + digit;
    }

    std::vector<unsigned int> result;
    while (number)
    {
        result.push_back(number % to_base);
        number /= to_base;
    }

    std::reverse(result.begin(), result.end());

    return result;
}
} // namespace all_your_base
